utilities/cli/container.py

```diff
@@ -439,10 +446,45 @@ def build(
         if full_build_args:
             cmd.extend(shlex.split(full_build_args))
 
-        cmd.extend(["-f", str(docker_file_path), "-t", img, str(HoloHubContainer.HOLOHUB_ROOT)])
+        cmd.extend(
+            [
+                "-f",
+                str(docker_file_path),
+                "-t",
+                img,
+                *(["-t", f"{img}-base"] if build_scripts else []),
+                str(HoloHubContainer.HOLOHUB_ROOT),
+            ]
+        )
 
         run_command(cmd, dry_run=self.dryrun)
 
+        if build_scripts:
+            for script in build_scripts:
+                script_path = get_holohub_setup_scripts_dir() / f"{script}.sh"
+                if not script_path.exists():
+                    fatal(f"Script {script}.sh not found in {get_holohub_setup_scripts_dir()}")
+                relative_script_path = script_path.relative_to(HoloHubContainer.HOLOHUB_ROOT)
+                cmd = [
+                    self.DOCKER_EXE,
+                    "build",
+                    "--build-arg",
+                    "BUILDKIT_INLINE_CACHE=1",
+                    "--build-arg",
+                    f"BASE_IMAGE={img}",
```

Comment:
I thought we aligned on some processes being based on the app img (ex: benchmark), some not (ex: lint) and unrelated to app/operator build